src/components/ArticleById.jsx

```diff
@@ -1,25 +1,31 @@
 import { useParams } from "react-router-dom";
 import { useEffect, useState } from "react";
-import { getArticleById } from "../utils/api";
+import { getArticleById, getCommentsOfArticle } from "../utils/api";
+import Comments from "./Comments";
 
 export default function ArticleById() {
   const [article, setArticle] = useState([]);
+  const [comments, setComments] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
   const { article_id } = useParams();
 
-
   useEffect(() => {
     setIsLoading(true);
-    getArticleById(article_id).then((res) => {
-      setArticle(res[0]);
-      setIsLoading(false);
+    Promise.all([
+      getArticleById(article_id),
+      getCommentsOfArticle(article_id),
+    ]).then((results) => {
+      setArticle(results[0][0]);
+      setComments(results[1]);
     });
```

Comment:
Nice implementation of the loading state here. 

I'd consider getting hold of this information in the request before sending it back to the component so that the api.js is acting more like a model accessing and formatting the data.